### PR TITLE
feat: Expand tones for less one-offs, fix `SUCCESS` a11y

### LINKED ISSUE #305: Button: Fix `SUCCESS` tone color contrast
`<Button tone="SUCCESS">` currently fails the WCAG 2 AA color contrast ratio:

image

Let's fix this analog to what we're already doing for `<Notification tone="SUCCESS">` and `<Badge tone="SUCCESS">`, and get rid of the following conditional styles by adjusting the "SUCCESS" `tone` accordingly:

- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Badge/Badge.helpers.tsx#L29-L31
- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Badge/Badge.helpers.tsx#L14-L16
- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Notification/Notification.helpers.ts#L55-L57

### LINKED ISSUE #309: Expand theme.tones to better support `WARNING`, <Notification variant="SOLID"> and similar
We currently provide the following [tones](https://gatsby-interface.netlify.app/?path=/story/theme-scales--tones), each (predominantly) picking colors from one color scale:

- `BRAND` — picking from the `purple` color scale
- `DANGER` — `red` color scale
- `NEUTRAL` — `grey` color scale
- `SUCCESS` — `green` color scale
- `WARNING` — `yellow` and `orange` color scale

Each `tone` currently provides the following colors: https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/theme/tones.ts#L5-L11

These tones are used to define

- border, background, text, and icon colors for [`Badge`](https://gatsby-interface.netlify.app/?path=/story/badge--sandbox), [`Notification`](https://gatsby-interface.netlify.app/?path=/story/notification--sandbox), and [`Button`](https://gatsby-interface.netlify.app/?path=/story/buttons-button--tones) and their `variant`s
- color for [`Heading`](https://gatsby-interface.netlify.app/?path=/story/heading--tones) and [`Text`](https://gatsby-interface.netlify.app/?path=/story/text--tones) 

Not all tones are available for all of the components listed above — the following components only support a subset of the available tones:

- [`Text`](https://gatsby-interface.netlify.app/?path=/story/text--tones) — `NEUTRAL`, `BRAND` 
- [`Heading`](https://gatsby-interface.netlify.app/?path=/story/heading--tones) and [`Button`](https://gatsby-interface.netlify.app/?path=/story/buttons-button--tones) — `BRAND`, `NEUTRAL`, `DANGER`, and `SUCCESS`

Note how `WARNING` isn't available for all of the aforementioned components?
It is, and always was available for `Notification` and its formerly two variants:

Bildschirmfoto 2020-04-27 um 00 12 17

These two "dark text on light background" variants worked _relatively_ well with the original `tone` scale. However we then introduced the [`SOLID` variant of `Notification`](https://gatsby-interface.netlify.app/?path=/story/notification--variants), and eventually expanded [`Badge`](https://gatsby-interface.netlify.app/?path=/story/badge--sandbox) to support all tones:

Bildschirmfoto 2020-04-27 um 00 13 36

Bildschirmfoto 2020-04-27 um 00 20 12

As the above screenshot shows, `WARNING` is the odd one out, requiring a bespoke text color for the `SOLID` variant.

We currently handle the text color requirements of `SOLID` `WARNING` via conditions — let's get rid of the following by expanding the `tone` variables to support a different text color for "solid" backgrounds:

https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Badge/Badge.tsx#L76-L79
- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Badge/Badge.helpers.tsx#L14-L16
- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Badge/Badge.helpers.tsx#L25-L28
- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Notification/Notification.tsx#L92-L95
- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Notification/Notification.tsx#L112-L115
- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Notification/Notification.tsx#L197-L203
- https://github.com/gatsby-inc/gatsby-interface/blob/665648fb8ae0a8aeaffda912dcaa47b2bc1a05ab/src/components/Notification/Notification.helpers.ts#L52-L54

### PR DESCRIPTION
**Pull request by fk**
This PR adds three new … tones ;) to the `tone` scale:

- `mediumInverted`
- `text`
- `textInverted`

These allow us to define the text and icon color of a `variant="SOLID"` ("inverted") component via `textInverted` and `mediumInverted`.  and also separate the text color definition from the default "dark text on light background".

The new `text` facilitates decoupling the existing `dark`, `darker`, and `superDark` colors from defining the `text` color of a component, allowing for more, and subsequently fixing the `SUCCESS` text color contrast issues.

https://deploy-preview-311--gatsby-interface.netlify.app/?path=/story/*

---

- Fix tones for `WARNING` and `SUCCESS`, and get rid of repeated conditions for one-off styles in components.
- I chose to not alter the existing variable names, or add new dedicated variables for `hover` states for now to keep this PR relatively easy to review.
- Fix `SUCCESS` tone color contrast — fixes #305  
  image
- Fixes #309 — allows removing all conditional styles for `WARNING` and `SUCCESS`.

Change overview:

- Add comments in `theme/tones.ts` about current usage.
- `Badge.STATUS.color = dark | superDark (for SUCCESS/WARNING)` -> `text`
- `Badge.PILL.color = white` -> `white` or `textInverted`, if the latter is defined
- `ConcealedValue.input.color = NEUTRAL.dark` -> `NEUTRAL.text` — makes that a slight bit darker. IMO works fine. Not sure if we should pull from tone here at all, but that's a question for another day.
- `Heading.color = superDark` -> `text`
  - Don't use a bespoke `tone` (`tone.dark`) for `Heading.LIGHT.color`, but stick to the default `tone.text` instead. Removing this distinction makes things a bit easier. Also I need to check where `LIGHT` is in use still, and think what to do with it — this looks a bit outdated.
- `Notification.SOLID.color = white` -> `white` or `textInverted`, if the latter is defined
- Refactor `Notification` a bit:
  - Let `NotificationContent.color` inherit the `baseCSS` or `variant` `color`, and remove `NotificationContent.color = NEUTRAL.superDark`. Instead add `color: theme.colors.grey[90]` to `baseCSS`. This makes me itch a little, but it's OK for now.
- `Notification.SOLID.Icon = whiteFade[90]` -> `whiteFade[90]` or `mediumInverted`, if the latter is defined
- `Text.color = darker` -> `text`
- `Button.[SECONDARY | GHOST].color = dark` -> `text`
- `Button.[SECONDARY | GHOST].border = dark` -> `superDark`
- Darken `NEUTRAL` tones:
  - With this change, `NEUTRAL.text` can be used for the `Text` and `Heading` components.
  - This subsequently makes `NEUTRAL` components more pronounced — specifically their text color (most notably the default `Text` color), and the `SOLID` variant's `background`.
  - This isn't a bad thing and has relatively low impact on `Notification` and `Badge` — in regards to `Button`, this is a good thing, because it actually improves its `disabled` state.
  - It also makes the `hover` state for the `NEUTRAL` button more pronounced, aligning it with its sibling `tone`s.
  - It's probably worth thinking about decoupling `Text` and `Heading` from the `NEUTRAL` tone.

### CODE REVIEW

**Comment by fk** [comment]
Please give me a bit more time, need to think about the `Text` default color change (which currently would change to be quite a bit darker than before).

**Comment by fk** [comment]
☝️ Done, restored original `<Text>` color.
This is 100% GTG now IMO.

File: src/components/Text/Text.helpers.ts [hunk 1]
```diff
@@ -3,7 +3,8 @@ import { ThemeCss } from "../../theme"
 
 export const baseStyle = (tone: TextTone): ThemeCss => {
   return theme => ({
-    color: theme.tones[tone].darker,
+    // TODO figure out a better way to do this
```
Comment (Elanhant) [inline src/components/Text/Text.helpers.ts:6]: If you don't mind, could you please add an issue for this and put its link in this comment?

Comment (fk) [reply src/components/Text/Text.helpers.ts:6]: 💯 Yes of course, will do!

**Pull request merged**